### PR TITLE
Pull request and issue templates

### PR DESCRIPTION
**Pull request by z64**
This PR adds two issue templates for bug reports and feature requests, as well as one for pull requests.

Low quality issues or PRs aren't really an issue we have - except sometimes we get questions best suited for the Discord channel (perhaps we should add a template for questions?). However for a lot of our contributors it's their first time contributing to an open source project, and some basic guidelines would help increase their confidence in contributing or reporting issues, and to reduce any up front questioning in the case of the latter.

The templates are intentionally sparse of structure for the most part, and hidden HTML comments (these do *not* show up in the rendered issue/pr) inform the author of the important points to hit in each respective case.

Feedback on these would be great!

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  Include why you want this feature. For example:
+
+  - It is a feature supported by Discord's API, but missing from the library
+
+  - It is a feature that would simplify a lot of your bots code
+
+  - It is a feature present in other popular Discord libraries
+
+  Include simple code examples (psuedocode is fine) that demonstrate
```
Comment (greenbigfrog) [inline .github/ISSUE_TEMPLATE/feature_request.md]: `pseudo`

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+---
+name: Bug report
+about: Create a report to help us improve
```
Comment (connorshea) [inline .github/ISSUE_TEMPLATE/bug_report.md]: Change to “Report a bug to help us improve the library”? Right now it feels like an incomplete sentence.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  Include simple code examples that will reproduce the bug.
+  Include any exceptions you are encountering in your logs, including
+  the initial error message and the backtrace that follows.
+
+  Stuck or need help? Join the Discord! https://discord.gg/cyK3Hjm
+-->
+
+---
+
+## Environment
```
Comment (connorshea) [inline .github/ISSUE_TEMPLATE/bug_report.md]: IMO this should include a section for the OS and version (e.g. `Ubuntu 18.04`, `Windows 10 1804`, or `macOS Mojave 10.14.0`)

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+<!---
+  These are some commands to run to give us basic information about
+  your Ruby environment. Some issues may be version, OS, or hardware specific.
+--->
+
+**Ruby version:**
+
+<!--- Paste full output of `$ ruby -v` here --->
```
Comment (connorshea) [inline .github/ISSUE_TEMPLATE/bug_report.md]: Remove the $ from these commands, they’ll probably just confuse noobs and are unnecessary for experienced people.

**Comment by z64** [comment]
Thank you @greenbigfrog , @kandayo , @connorshea 👍

**Pull request merged**